test/util/core/deep-equal-test.js

```diff
@@ -352,4 +352,51 @@ describe("util/core/deepEqual", function () {
         assert(matchDeepEqual(matchA, duplicateA));
         assert.isFalse(matchDeepEqual(matchA, matchB));
     });
+
+    it("handles shallow cyclic objects", function () {
+        var a = {
+            foo: "bar"
+        };
+        a.cyclicKeyName = a;
+
+        var b = {
+            foo: "bar"
+        };
+        b.cyclicKeyName = b;
+
+        assert(deepEqual(a, b));
+    });
+
+    it("handles deep cyclic objects", function () {
+        var a = {
+            foo: "bar",
+            key: { }
+        };
+        a.key.value = a;
+
+        var b = {
+            foo: "bar",
+            key: { }
+        };
+        b.key.value = b;
+
+        assert(deepEqual(a, b));
+    });
+
+    it.skip("handles cyclic objects when a matcher provided", function () {
```

Comment:
Is this intentional? You meant to add a skipped test?

Reply:
Added a skipped tests is weird to me. It sounds like this test needs to be added when the bug is fixed or you need to assert that the functionality is broken as expected. Adding a skipped test is like adding a commented out test to me. It doesn't do anything meaningful for the current code base.